### ISSUE TITLE
Parameter epoch_size greater than dataset size in train phase

### ISSUE THREAD

**Issue by murilovarges** [facebookresearch/R2Plus1D#9]
Hi,

In the paper, you said _"Although Kinetics has only about 240k training videos, we set epoch size to be 1M for temporal jittering"_. 

I would like to know what will happen, the video reader will start from the begin of dataset? In this case will process the dataset about 4 times per epoch, in each time get a different clip for each video using temporal jittering?

I tried to find this information reading source codes in the folder pytorch/caffe2/video/*.* but I remain with doubt.

Thank's Murilo.

**Comment by dutran** [comment]
Yes, it uses temporal jitter, which is implemented here https://github.com/pytorch/pytorch/blob/master/caffe2/video/video_input_op.h#L235

**Comment by murilovarges** [comment]
Thank you @dutran, 

In https://github.com/pytorch/pytorch/blob/master/caffe2/core/db.h#L232 If the cursor reaches its end, the reader will go back to the head of the db.

**Comment by dutran** [comment]
Yes, the reader will make about 4 times over the list, but due to temporal jittering, it is likely to see a different clip even with the same video.

**Comment by cr-hxfan** [comment]
Hi, @dutran   May I know how much the accuracy improved by temporal jittering?  Did you test it before?
Thanks in advance.

**Comment by cr-hxfan** [comment]
@dutran Also, it seems like your code in GitHub didn't use temporal jitter?

In the model helper, you set the "clip_per_video" as 1:
https://github.com/facebookresearch/R2Plus1D/blob/54d93222f80cd23a87c7dbda6044b3fae07b3fda/tools/train_net.py#L249

When it is 1, it seems like the "clip_stat" is alway 0, which can be seen from the source code of caffe2,
https://github.com/pytorch/pytorch/blob/ca03c10cefa1e126eab1446d490f9314bd236c1b/caffe2/video/video_io.cc#L538

If I am correct, when you set "clip_per_video" as 1, you will always use the same clip for each video?

@murilovarges What do you think?  Am I right?
